File: TradeItIosEmsApi/TradeItQuotesRequest.m
//
//  TradeItQuotesRequest.m
//  TradeItIosEmsApi
//
//  Created by Antonio Reyes on 2/12/16.
//  Copyright © 2016 TradeIt. All rights reserved.
//

#import "TradeItQuotesRequest.h"

@implementation TradeItQuotesRequest

-(id) initWithSymbol:(NSString *) symbol
           andApiKey:(NSString *) apiKey {
    self = [super init];
    if(self) {
        self.symbol = symbol;
        self.apiKey = apiKey;
    }
    
    if([symbol containsString:@".SI"]) {
        self.symbol = [symbol substringToIndex:[symbol rangeOfString:@".SI"].location];
        self.suffixMarket = @"SI";
    }
    
    return self;
}

-(id) initWithFxSymbol:(NSString *) symbol
             andBroker:(NSString *)broker
             andApiKey:(NSString *) apiKey {
    self = [self initWithSymbol:symbol andApiKey:apiKey];
    self.broker = broker;
    return self;
}

-(id) initWithSymbols:(NSArray *) symbols
            andApiKey:(NSString *) apiKey {
    self = [super init];
    if(self) {
        self.symbols = [symbols componentsJoinedByString:@","];
        self.apiKey = apiKey;
    }
    return self;
}

-(id) initWithSymbol:(NSString *) symbol
     andMarketSuffix: (NSString *) suffix
           andApiKey:(NSString *) apiKey {
    self = [super init];
    if(self) {
        self.symbol = symbol;
        self.suffixMarket = suffix;
        self.apiKey = apiKey;
    }
    return self;
}

@end

File: TradeItIosEmsApi/TradeItQuotesRequest.swift
class TradeItQuotesRequest: TradeItRequest, Codable {
    
    var symbol: String?
    var symbols: String?
    var broker: String?
    var apiKey: String
    var suffixMarket: String?
    
    init(symbol: String, apiKey: String) {
        self.symbol = symbol
        self.apiKey = apiKey
    }
    
    init(symbols: [String], apiKey: String) {
        self.symbols = symbols.joined(separator: ",")
        self.apiKey = apiKey
    }
    
    init(symbol: String, broker: String, apiKey: String) {
        self.symbol = symbol
        self.broker = broker
        self.apiKey = apiKey
    }
    
    init(symbol: String, suffixMarket: String, apiKey: String) {
        self.symbol = symbol
        self.suffixMarket = suffixMarket
        self.apiKey = apiKey
    }
        
}
